stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/utils/CustomObjectFiltering.kt

```diff
@@ -91,3 +91,18 @@ private fun <T : Comparable<T>> compare(a: T?, b: T?, compareFun: (Int) -> Boole
 
 private fun CustomObject.getMembersId(): List<String> =
     getMemberPropertyOrExtra(MEMBERS_FIELD_NAME, List::class)?.mapNotNull { (it as? Member)?.getUserId() } ?: emptyList()
+
+@Suppress("UNCHECKED_CAST")
+private fun <T : Any> CustomObject.getExtra(name: String, clazz: KClass<out T>): T? =
+    extraData[name]?.let {
+        when (clazz) {
```

Comment:
Was not sure about comparisons of numbers with floating points as such operations are not precise. But seems to work fine.